main.tf

```
@@ -177,8 +177,8 @@ resource "azurerm_windows_function_app" "this" {
   service_plan_id     = var.app_service_plan_id
 
   storage_account_name          = local.storage_account_name
-  storage_account_access_key    = null # Use managed identity instead
-  storage_uses_managed_identity = true
+  storage_account_access_key    = !var.storage_uses_managed_identity ? var.storage_account_access_key : null # Use managed identities instead
```

This comment doesn't really make sense now that we no longer enforce identity-based authentication:

```suggestion
  storage_account_access_key    = !var.storage_uses_managed_identity ? var.storage_account_access_key : null
```